lib/Core/AsyncLoader.ts

```diff
@@ -85,7 +90,11 @@ export default class AsyncLoader {
     return this._isLoading;
   }
 
-  load(forceReload: boolean = false): Promise<void> {
+  get result() {
+    return this._result;
+  }
+
+  async load(forceReload: boolean = false): Promise<Result<void>> {
```

Comment:
a suggestion would be to return a `Promise<Result<T>>` where T is a type of loadCallback function passed in to the constructor ie `loadCallback: () => Promise<T>`. This way, we can do things like `const result = (await this.loadMetadata())`. Also i couldn't find any usages of the disposeCallback function.

Reply:
Yeah I do like this idea, but getting the types to work is a pain - also `AsyncLoader` usage varies quite a lot. So this approach would work for some implementations (eg GeoJSON), but not for others (eg TableMixin and AutoRefreshMixin).

As some point we need to re-think the `AsyncLoader` approach. @steve9164 and I have spent a bit of time thinking about it over the past year or so - but we decided it wasn't worth re-designing for v8.